rental.rb

```diff
@@ -1,22 +1,49 @@
+require_relative './person'
+
 class Rental
-  attr_accessor :date
-  attr_reader :book, :person
+  attr_accessor :date, :person, :book
 
   def initialize(date, book, person)
     @date = date
     @book = book
     book.rentals << self
     @person = person
-    person.rental << self
+    person.rentals << self
   end
 
-  def book=(book)
-    @book = book
-    book.rentals.push(self) unless book.rentals.include?(self)
+  def self.create_rental(books, people)
+    if books.empty?
+      puts 'Please create a book first (option 4).'
+    elsif people.empty?
+      puts 'Please add a person first (option 3).'
+    else
+      puts 'Select a book from this list by number'
+      books.each do |book|
+        puts "#{books.find_index(book)} - #{book.title}"
+      end
+      selected_book = gets.to_i
+      print 'Date: '
+      date = gets.chomp
+      puts 'Select a person from this list by number'
+      people.each do |person|
+        puts "#{people.find_index(person)} - #{person.name}"
+      end
+      selected_person = gets.to_i
+
+      Rental.new(date, books[selected_book], people[selected_person])
+    end
   end
 
-  def person=(person)
-    @person = person
-    person.rentals.push(self) unless person.rentals.include?(self)
+  def self.list_all_rentals_person_id(people, rentals)
+    Person.list_all_people(people)
+    print 'Person ID: '
+    person_id = gets.to_i
+    puts(rentals.map do |rental|
+      if rental.person.id == person_id
+        "Rentals:\nDate: #{rental.date}  Book: #{rental.book.title}"
+      else
+        'No rentals found for this person'
+      end
+    end)
   end
 end
```

Comment:
- Hey, good job so far! Here's the error I get when I try to run the app. Kindly check your Rentals because the error is preventing the app from running. See the image below:

*Run error*
![Screenshot from 2023-02-03 10-02-58](https://user-images.githubusercontent.com/59397159/216557965-19637a85-502b-4020-98f1-6ec74f1ba6a5.png)

Reply:
### README FILE

- Please, let's remove these placeholder texts as these texts are meant to guide on how to update your readme with your content and are not expected to be in the readme. please, let's remove them to make our readme more professional.



![Screenshot from 2023-02-03 10-07-14](https://user-images.githubusercontent.com/59397159/216558898-1a1db362-f132-4303-840d-5af6dc61557a.png)
![Screenshot from 2023-02-03 10-07-06](https://user-images.githubusercontent.com/59397159/216558916-5aef12a4-677f-4fc3-8c4a-b7376dbba366.png)
![Screenshot from 2023-02-03 10-06-58](https://user-images.githubusercontent.com/59397159/216558918-0874917e-8377-423f-9201-a32212abf301.png)
![Screenshot from 2023-02-03 10-06-50](https://user-images.githubusercontent.com/59397159/216558920-09bd20de-e333-44cd-a55c-1e504c63cc1b.png)
![Screenshot from 2023-02-03 10-06-42](https://user-images.githubusercontent.com/59397159/216558926-a8c5d8da-4a53-437d-b5bb-67dc8c176a20.png)
![Screenshot from 2023-02-03 10-06-31](https://user-images.githubusercontent.com/59397159/216558931-32114723-1964-4aec-9589-0c7b424f88dc.png)
![Screenshot from 2023-02-03 10-06-25](https://user-images.githubusercontent.com/59397159/216558934-5b725356-2c2a-4fb8-9f1d-c1b0da89f76e.png)

Reply:
> * Hey, good job so far! Here's the error I get when I try to run the app. Kindly check your Rentals because the error is preventing the app from running. See the image below:
> 
> _Run error_ ![Screenshot from 2023-02-03 10-02-58](https://user-images.githubusercontent.com/59397159/216557965-19637a85-502b-4020-98f1-6ec74f1ba6a5.png)

Thanks, @zieeco, but actually, the app is running fine here on my system... I did not have such an error...
Do you have a clue on what the reason might be